examples/optimize_evoked.py

```diff
@@ -0,0 +1,433 @@
+        combined_chunk['ranges'].update(evinput['ranges'])
+        if evinput['opt_end'] > combined_chunk['opt_end']:
+            combined_chunk['opt_end'] = evinput['opt_end']
+
+    # wRMSE with weights of 1's is the same as regular RMSE.
+    combined_chunk['weights'] = np.ones(len(input_chunks[-1]['weights']))
+    combined_chunk['num_sims'] = default_num_total_sims
+
+    return combined_chunk
+
+def consolidate_chunks(inputs):
+    """ Consolidates inputs into optimization "chunks" defined by
+    opt_start and opt_end
+    Parameters
+    ----------
+    inputs: Dict
+        Sorted dictionary of inputs with their parameters
+        and weight functions
+
+    Returns
+    -------
+    consolidated_chunks: List
+        List of chunks corresponding to optimization steps
+    """
+    global default_num_step_sims
+
+    consolidated_chunks = []
+    for input_name in inputs.keys():
+        input_dict = inputs[input_name].copy()
+        input_dict['inputs'] = [input_name]
+        input_dict['num_sims'] = default_num_step_sims
+
+        if (len(consolidated_chunks) > 0) and \
+            (input_dict['start'] <= consolidated_chunks[-1]['end']):
+            # update previous chunk
+            consolidated_chunks[-1]['inputs'].extend(input_dict['inputs'])
+            consolidated_chunks[-1]['num_params'] += input_dict['num_params']
+            consolidated_chunks[-1]['end'] = input_dict['end']
+            consolidated_chunks[-1]['ranges'].update(input_dict['ranges'])
+            consolidated_chunks[-1]['opt_end'] = max(consolidated_chunks[-1]['opt_end'], input_dict['opt_end'])
+            # average the weights
+            consolidated_chunks[-1]['weights'] = (consolidated_chunks[-1]['weights'] + input_dict['weights'])/2
+        else:
+            # new chunk
+            consolidated_chunks.append(input_dict)
+
+    # add one last chunk to the end
+    if len(consolidated_chunks) > 1:
+        last_chunk = create_last_chunk(consolidated_chunks)
+        consolidated_chunks.append(last_chunk)
+
+    return consolidated_chunks
+
+def optrun (new_params, grad=0):
+    """ This is the function to run a simulation
+    Parameters
+    ----------
+    new_params: Array
+        List or numpy array with the parameters chosen by
+        optimization engine. Order is consistent with
+        opt_params['ranges']
+    grad: float | None
+        Gradient if provided by optimization algorithm
+
+    Returns
+    -------
+    avg_rmse: float
+        Weighted RMSE between data in dpl and exp_dpl
+    """
+    global opt_params, params, exp_dpl, best_dpl, cur_step, optiter, stepminopterr
+    print("Optimization step %d, iteration %d"%(cur_step+1, optiter+1))
+
+    for param_name, test_value in zip(opt_params['ranges'].keys(), new_params): # set parameters
+        if test_value >= opt_params['ranges'][param_name]['minval'] and test_value <= opt_params['ranges'][param_name]['maxval']:
```

Comment:
@blakecaldwell would you mind running a flake8 linter on the code? It makes it a bit easier to read the code

Reply:
Yeah, no problem for the future. I have limited time to finish this request that you made before the MNE sprint. Figured you'd still be able to do something with this code as-is.